python_modules/dagster/dagster/_config/pythonic_config/typing_utils.py

```diff
@@ -51,7 +55,28 @@ def set_actual_types_for_type_checking(
 
 
 @dataclass_transform(kw_only_default=True, field_specifiers=(Field,))
-class BaseResourceMeta(pydantic.main.ModelMetaclass):
+class BaseConfigMeta(pydantic.main.ModelMetaclass):
+    def __new__(cls, name, bases, namespaces, **kwargs) -> Any:
+        annotations = namespaces.get("__annotations__", {})
+        for field in annotations:
+            if not field.startswith("__"):
+                # Need try/catch because DagsterType may not be loaded when some of the base Config classes are
+                # being created
+                # Any user-created Config class will have DagsterType loaded by the time it's created, so this
+                # will only affect the base Config classes (where this error won't be an issue)
+                try:
+                    from dagster._core.types.dagster_type import DagsterType
+
+                    if isinstance(annotations[field], DagsterType):
```

Comment:
what does `annotations[field]` contain in the case where DagsterType is not loaded?

Reply:
`annotations[field]` is the type annotation of the field on the `Config` class. 

```python
class MyOpConfig(Config):
    foo: str
    bar: List[int]
```

In this case, `str` or `List[int]`.

In the case that `DagsterType` is not yet available, it means whatever annotation is in use must not be a `DagsterType` (e.g. a primitive `str`, `int`, another `Config` class etc).